azure-quantum/tests/unit/test_qiskit.py

```diff
@@ -537,3 +537,82 @@ def test_qiskit_get_rigetti_qpu_targets(self):
         assert "rigetti" == config.azure["provider_id"]
         assert "qir.v1" == config.azure["input_data_format"]
         assert "microsoft.quantum-results.v1" == config.azure["output_data_format"]
+
+    @pytest.mark.qci
+    @pytest.mark.live_test
+    def test_qiskit_submit_to_qci(self):
+        with unittest.mock.patch.object(
+            Job,
+            self.mock_create_job_id_name,
+            return_value=self.get_test_job_id(),
+        ):
+            workspace = self.create_workspace()
+            provider = AzureQuantumProvider(workspace=workspace)
+            assert "azure-quantum-qiskit" in provider._workspace.user_agent
+            backend = provider.get_backend("qci.simulator")
+            assert backend.name() == "qci.simulator"
+            config = backend.configuration()
+            assert True == config.simulator
+            assert 1 == config.max_experiments
+            assert 29 == config.num_qubits
+            assert "qir.v1" == config.azure["content_type"]
+            assert "qci" == config.azure["provider_id"]
+            assert "qir.v1" == config.azure["input_data_format"]
+            assert "microsoft.quantum-results.v1" == config.azure["output_data_format"]
+            shots = 100
+
+            circuit = self._3_qubit_ghz()
+            circuit.metadata = { "some": "data" }
+
+            qiskit_job = backend.run(
+                circuit=circuit,
+                shots=shots
+            )
+
+            # Check job metadata:
+            assert qiskit_job._azure_job.details.target == "qci.simulator"
+            assert qiskit_job._azure_job.details.provider_id == "qci"
+            assert qiskit_job._azure_job.details.input_data_format == "qir.v1"
+            assert qiskit_job._azure_job.details.output_data_format == "microsoft.quantum-results.v1"
+            assert "qiskit" in qiskit_job._azure_job.details.metadata
+            assert "name" in qiskit_job._azure_job.details.metadata
+            assert "num_qubits" in qiskit_job._azure_job.details.metadata
+            assert "metadata" in qiskit_job._azure_job.details.metadata
+            assert qiskit_job._azure_job.details.metadata["num_qubits"] == '4'
+            assert qiskit_job._azure_job.details.input_params["count"] == shots
+            assert qiskit_job._azure_job.details.input_params["entryPoint"] == "main"
+            assert qiskit_job._azure_job.details.input_params["arguments"] == []
+
+            # Make sure the job is completed before fetching the results
+            self._qiskit_wait_to_complete(qiskit_job, provider)
+
+            if JobStatus.DONE == qiskit_job.status():
+                result = qiskit_job.result()
+                print(result)
+                assert sum(result.data()["counts"].values()) == shots
+                assert np.isclose(result.data()["counts"]["[0, 0, 0]"], shots//2, 20)
```

Comment:
Question out of curiosity, why are we not using regular division in `shots/2`?

Reply:
it's unclear to me too :)